core/src/main/scala/kafka/api/ApiVersion.scala

```diff
@@ -94,19 +94,19 @@ object ApiVersion {
     versionNameMap.values.toSet
   }
 
-  def minVersionForMessageFormat(messageFormatVersion: RecordFormat): String = {
-    messageFormatVersion match {
-      case RecordFormat.V0 => "0.8.0"
-      case RecordFormat.V1 => "0.10.0"
-      case RecordFormat.V2 => "0.11.0"
-      case _ => throw new IllegalArgumentException(s"Invalid message format version $messageFormatVersion")
+  def minVersionForRecordVersion(recordVersion: RecordVersion): String = {
```

Comment:
nit: the redundancy in this name is slightly vexing. Maybe we can leave off `ForRecordVersion` since it's sort of implicit from the argument anyway. Perhaps `minSupportingVersion`?